--- setup.py
+++ setup.py
@@ -109,7 +109,5 @@
 ]
 extra_compile_args = [
-    '-O0',
-    '-ftemplate-depth-256',
-    '-Wno-unused-function',
-    '-Wno-uninitialized',
+    '/EHsc',
+    '-DBOOST_DATE_TIME_NO_LIB'
 ]

--- pystan/model.py
+++ pystan/model.py
@@ -289,7 +289,5 @@
        ]
        extra_compile_args = [
-            '-O0',
-            '-ftemplate-depth-256',
-            '-Wno-unused-function',
-            '-Wno-uninitialized',
+            '/EHsc',
+            '-DBOOST_DATE_TIME_NO_LIB'
        ]
